server/src/main/java/org/apache/druid/segment/realtime/appenderator/BatchAppenderator.java

```diff
@@ -785,6 +785,9 @@ private DataSegment mergeAndPush(
         for (FireHydrant fireHydrant : sink) {
           Pair<ReferenceCountingSegment, Closeable> segmentAndCloseable = fireHydrant.getAndIncrementSegment();
           final QueryableIndex queryableIndex = segmentAndCloseable.lhs.asQueryableIndex();
+          if (queryableIndex != null) {
```

Comment:
Why would the queryable index by empty here. 
Would it cause problems in the indexes.add() method ?

Reply:
I think the getter above would return null if the queryable index is closed. I think we gracefully handle this elsewhere, I haven't changed that part of the code. This is just to avoid throwing an NPE here if we do run into this problem, since we are just reporting metrics here.